libs/sr/ui/src/LightCone/LightConeEditor/LightConeEditor.tsx

```diff
@@ -0,0 +1,342 @@
+  const sheet: ILightConeSheet | undefined = lightCone
+    ? tempLightConeSheet
+    : undefined
+
+  const update = useCallback(
+    (newValue: Partial<ILightCone>) => {
+      // const newSheet = newValue.key ? getLightConeSheet(newValue.key) : sheet!
+      const newSheet = newValue.key ? tempLightConeSheet : sheet!
+
+      function pick<T>(
+        value: T | undefined,
+        available: readonly T[],
+        prefer?: T
+      ): T {
+        return value && available.includes(value)
+          ? value
+          : prefer ?? available[0]
+      }
+
+      if (newValue.key) {
+        newValue.superimpose = pick(
+          lightCone?.superimpose,
+          newSheet.superimpose,
+          Math.min(...newSheet.superimpose) as SuperimposeKey
+        )
+      }
+      if (newValue.level) {
+        newValue.level = clamp(newValue.level, 0, lightConeMaxLevel)
+      }
+
+      dispatchLightCone({ type: 'update', lightCone: newValue })
+    },
+    [lightCone, sheet, dispatchLightCone]
+  )
+
+  const reset = useCallback(() => {
+    dispatchLightCone({ type: 'reset' })
+  }, [])
+
+  useEffect(() => {
+    if (lightConeIdToEdit === 'new') {
+      dispatchLightCone({ type: 'reset' })
+    }
+  }, [lightConeIdToEdit])
+
+  return (
+    <Container>
+      <CardThemed bgt="dark">
+        <CardHeader title="Light Cone Editor" />
+        <CardContent sx={{ display: 'flex', flexDirection: 'column', gap: 1 }}>
+          <Grid container spacing={1} columns={{ xs: 1, md: 2 }}>
+            {/* name */}
+            <Grid item xs={1} md={true} display="flex" flexDirection="column">
+              <LightConeAutocomplete
+                lcKey={lightCone?.key ?? ''}
+                setLCKey={(lcKey) => update({ key: lcKey as LightConeKey })}
+                label={lightCone?.key ? '' : t('editor.unknownLightCone')}
+              />
+            </Grid>
+
+            {/* superimpose */}
+            <Grid item xs={true} md="auto" display="flex">
+              <SuperimpositionDropdown
+                superimpose={lightCone?.superimpose}
+                setSuperimposition={(sk: SuperimposeKey) =>
+                  update({ superimpose: sk })
+                }
+                disabled={!lightCone}
+              />
+            </Grid>
+          </Grid>
+          <Grid
+            container
+            spacing={1}
+            columns={{ xs: 1, md: 4 }}
+            marginBottom={1}
+          >
+            {/* level */}
+            <Grid item xs={1} display="flex" flexDirection="row" gap={1}>
+              <LevelDropdown
+                level={lightCone?.level}
+                ascension={lightCone?.ascension}
+                setLevelAscension={(lv, as) => {
+                  update({ level: lv, ascension: as })
+                }}
+                disabled={!lightCone}
+              />
+            </Grid>
+
+            {/* ascension */}
+            <Grid item xs={1} display="flex" gap={1}>
+              <TextField
```

Comment:
Can you style this  more similar to the relic editor for their readonly fields, such as slot name? https://github.com/frzyc/genshin-optimizer/blob/181f49632e641c01061b1f4b2b7a5675a21212e1/libs/sr/ui/src/Relic/RelicEditor/RelicEditor.tsx#L244